src/features/slackapi/slackapi-utils.ts

```diff
@@ -107,30 +146,42 @@ Have a great week!
    * @param previousWorkDays dates and the number of today
    * @param nonProjectTimes all non project times
    */
-  export const postDailyMessage = (
+  export const postDailyMessage = async (
     dailyCombinedData: DailyCombinedData[],
     timeRegistrations: TimeRegistrations[],
     previousWorkDays: PreviousWorkdayDates,
-    nonProjectTimes: NonProjectTime[]) => {
-    const { numberOfToday, yesterday, today } = previousWorkDays;
-
-    dailyCombinedData.forEach(user => {
-      const { slackId, personId, expected } = user;
-
-      const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
-      const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);
-
-      if (!isAway && !firstDayBack) {
-        try {
-          client.chat.postMessage({
-            channel: slackId,
-            text: constructDailyMessage(user, numberOfToday)
-          });
-        } catch (error) {
-          console.error(`Error while posting slack messages to user ${user.name}`);
+    nonProjectTimes: NonProjectTime[]): Promise<DailyMessageData[]> => {
+    try{
+      const { numberOfToday, yesterday, today } = previousWorkDays;
+
+      let messagesRecord: DailyMessageData[] = [];
+
+      const promises = dailyCombinedData.map(user => {
+        const { slackId, personId, expected } = user;
+
+        const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
+        const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);
+
```

Comment:
```suggestion
```

Reply:
Should firstDayBack be removed along with the conditional right below it?